hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroKafkaSource.java

```diff
@@ -91,12 +96,21 @@ public AvroKafkaSource(TypedProperties props, JavaSparkContext sparkContext, Spa
       return new InputBatch<>(Option.empty(), CheckpointUtils.offsetsToStr(offsetRanges));
     }
     JavaRDD<GenericRecord> newDataRDD = toRDD(offsetRanges);
-    return new InputBatch<>(Option.of(newDataRDD), KafkaOffsetGen.CheckpointUtils.offsetsToStr(offsetRanges));
+    return new InputBatch<>(Option.of(newDataRDD), CheckpointUtils.offsetsToStr(offsetRanges));
   }
 
   private JavaRDD<GenericRecord> toRDD(OffsetRange[] offsetRanges) {
-    return KafkaUtils.createRDD(sparkContext, offsetGen.getKafkaParams(), offsetRanges,
-            LocationStrategies.PreferConsistent()).map(obj -> (GenericRecord) obj.value());
+    if (deserializerClassName.equals(ByteArrayDeserializer.class.getName())) {
+      if (schemaProvider == null) {
+        throw new HoodieException("Please provide a valid schema provider class!");
```

Comment:
Given more exception information, e.g. : when choose `ByteArrayDeserializer `?

Reply:
OK, good idea,'when use ByteArrayDeserializer ' would be better?